apps/slack/lambda/lib/routes/auth-token/repository.ts

```diff
@@ -47,24 +49,43 @@ export class AuthTokenRepository {
       accessResponse.team!.id!
     );
 
-    const authToken = AuthTokenRepository.toDatabase(accessResponse, {
-      spaceId,
-      environmentId,
-    });
+    const authToken = AuthTokenRepository.toDatabase(
+      accessResponse,
+      {
+        spaceId,
+        environmentId,
+      },
+      installationUuid
+    );
 
     await this.updateByWorkspaceId(authToken.slackWorkspaceId, authToken);
 
-    await this.singleTableClient.put(Entity.AuthToken, uuid, [spaceId, environmentId], authToken);
+    await this.singleTableClient.put(
+      Entity.AuthToken,
+      installationUuid || uuid,
+      [spaceId, installationUuid || environmentId],
+      authToken
+    );
 
     return authToken;
   }
 
   async get(workspaceId: string, context: SpaceEnvironmentContext): Promise<AuthToken> {
-    let authToken = await this.singleTableClient.get(
-      Entity.AuthToken,
-      AuthTokenRepository.uuid(context.spaceId, context.environmentId, workspaceId)
+    const parameters = await getInstallationParametersFromCma(
+      context.spaceId,
+      context.environmentId
+    );
+    const installationUuid = parameters.installationUuid;
+    const spaceEnvUuid = AuthTokenRepository.uuid(
+      context.spaceId,
+      context.environmentId,
+      workspaceId
     );
 
+    // installationUuid is preferred, spaceEnvUuid is for backwards compatibility
+    const possibleUuids = installationUuid || spaceEnvUuid;
```

Comment:
backwards compatibility flow here

Reply:
Super small nit but it confused me at first: Maybe rename to `possibleUuid` (singular) or even `uuidToUse` or something. Your current name made me at first think this was an array / collection of uuids.

Reply:
ah yes good point! Will adjust 👍